Package lascar.0.5

jserot opened the pull request:
### `lascar.0.5`
A library for manipulating Labeled Transition Systems in OCaml



---
* Homepage: http://cloud.ip.univ-bpclermont.fr/~serot/lascar/
* Source repo: git+https://github.com/jserot/lascar.git
* Bug tracker: jocelyn.serot@uca.fr

---
:camel: Pull-request generated by opam-publish v2.0.0

camelus commented:
##### :sunny: All lint checks passed <small>e8c251f89d9ff95bd2812b02c878f82a3efa53b9</small>

* These packages passed lint tests: `lascar.0.5`


---


##### :sunny: Installability check (10843 &rarr; 10843)

kit-ty-kate commented on packages/lascar/lascar.0.5/opam:
```diff
@@ -18,17 +16,3 @@ install: [
-  "ocamlfind" {build}
-]
-synopsis: "A library for manipulating Labeled Transition Systems in OCaml"
-url {
```
the url section is removed and thus doesn't do anything anymore

kit-ty-kate commented on packages/lascar/lascar.0.5/opam:
```diff
@@ -1,14 +1,12 @@
+homepage: "http://cloud.ip.univ-bpclermont.fr/~serot/lascar/"
 bug-reports: "jocelyn.serot@uca.fr"
+dev-repo: "git+https://github.com/jserot/lascar.git"
+depends: [ "camlp4" "ocamlfind" {build} ]
```
the previous depends section was more expressive (lower bound on ocaml) and more readable (indentation). Could you put it back?

mseri commented:
I think this is incorrectly modifying the existing version 0.5 instead of releasing a new 0.6 version

jserot commented:
> I think this is incorrectly modifying the existing version 0.5 instead of releasing a new 0.6 version

In fact, this is only a repackaging of 0.5 for opam2. 
Should i have retaged the version 0.6 ?

mseri commented:
Sorry, no then it's fine, but please add the fixes suggested by @kit-ty-kate

jserot commented:
Build fails with the following error message : 
`#=== ERROR while compiling lascar.0.5 =========================================#`
`"./configure": command not found.`

I've checked that he `configure` file script is present both on the local and git repos. 
Where does the opam build script fetch the source tree from ?

mseri commented:
It does not, this PR removes the url section: https://github.com/ocaml/opam-repository/pull/13701#discussion_r267079442

jserot commented:
Ah ok. I bring it back and retry then.

jserot commented:
Is the url section mandatory ?
The documentation seems to imply that it is only required when not_using opam publish...
I remember having said to myself : this means that, with opam publish the source tree is fetched from github and i don't have to provide a tarball anymore.. Am i wrong ?

mseri commented:
It is usually added by `opam-publish` to the opam file in the PR, I don't know why it is not happening. Maybe it cannot detect the right tag?

mseri commented:
Yes, there is no 0.5 or v0.5 tag on the github sources for lastcar, so opam-publish does not find anything to populate the url field.

jserot commented:
Ah ok. I must have pushed to github _before_ tagging :/
Will fix this. Thx.

jserot commented:
Btw, you confirm that having an URL location with a tar ball is non longer required ?

mseri commented:
You do need it in the published opam file, but it should be generated by opam-publish when preparing the PR

jserot commented:
Tagging the github repo does not seem to solve the problem.

mseri commented:
I have added the url section by hand

jserot commented:
Thanks !
Is the updated version that available [here](https://github.com/ocaml/opam-repository/blob/master/packages/lascar/lascar.0.5/opam) ?
If yes, does this mean that a link to an external tarball is still required for publishing ?

mseri commented:
It is the version that you tagged on github: https://github.com/ocaml/opam-repository/pull/13701/commits/59d596bc1e8bfac800f025206b14d89fc89b0825#diff-06241e92e07bb6b76aca9a21b35293d3R24

Yes, the published opam needs to link to a tarball

mseri commented:
The revdeps error is because one of the package seems unreachable:
```
[ERROR] The sources of the following couldn't be obtained, aborting:
          - rfsm.1.0:
            http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm.tar.gz (curl: code 404 while downloading http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm.tar.gz)
```

jserot commented:
> It is the version that you tagged on github: 59d596b#diff-06241e92e07bb6b76aca9a21b35293d3R24

Ok. IIUC, `github` creates this tar ball automatically when a a release is declared (tagged).

> The revdeps error is because one of the package seems unreachable

Fixed (the corresponding file had been renamed). Does this mean that for a package to be validated, all packages _depending on it_ have to be available (as the name "revdeps" seems to apply) ? This is a bit counterintuitive, isn't it  ?

mseri commented:
It is not compulsory that the revdeps are green. Sometimes they cannot succeed because some c libraries are not available for some specific distributions, for example. But among the tests, we check that each release can build the libraries it depends on, and in case of breaking changes breaking some dependencies and so on (when possible) we update the upper bounds on their opam files (and sometimes make a PR or notify the maintainers upstreams). In this way when people upgrade their opam environment or create new environments, the risk of breaking everything and remain in an inconsistent state is minimized.

mseri commented:
As an example of unavoidable failures: the camlp4 failure for system switches in Travis is expected and we cannot do much about that.

mseri commented:
Note that rfsm checksum has changed. We can fix that in a separate PR if you prefer, this one is fine for merging
```
[ERROR] The sources of the following couldn't be obtained, aborting:
          - rfsm.1.0:
            http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm.tar.gz (Bad checksum, expected md5=e4d7f1086eb346e7a61760036f059bc6)
```

jserot commented:
Thanks for the explanations. 

Concerning RFSM, the change in checksum is normal :  the tarball available on the `cloud.ip...` is for version 1.4 and my initial idea was to update the OPAM package _after_ LASCAR.

 IIUC, the best fix, now, would be to tag a new release of RFSM on github and have OPAM use this one instead on that available on `cloud.ip...`. Yes ?

mseri commented:
Is there an available tarball for version 1.0 that we could use for rfsm.1.0? We tend to keep old package version for people that are stuck with old compilers or old dependencies. Separately you can release rfsm.1.4 as a different package

jserot commented:
The oldest version i have is [here](http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm-source-1.3.tar.gz). I can't remember whether this is the one that the OPAM `rfsm` package was refering to.

mseri commented:
@rjbou do we have a copy of that in the opam cache that we could use for the url of rfsm.1.0?

kit-ty-kate commented:
@mseri yes we have, just take what's at `opam.ocaml.org/cache/md5/<two_first_char_of_the_hash>/<hash>`

mseri commented:
@jserot could you publish `https://opam.ocaml.org/cache/md5/e4/e4d7f1086eb346e7a61760036f059bc6` as `http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm.1.0.tar.gz` and update the url of `rfsm.1.0/opam` to point to it?

jserot commented:
@mseri 
> could you publish https://opam.ocaml.org/cache/md5/e4/e4d7f1086eb346e7a61760036f059bc6 as http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm.1.0.tar.gz 

Done

> and update the url of rfsm.1.0/opam to point to it?

There was no URL section in rfsm.1.0/opam (which was for opam 1.2)
Should i add one ? But this will change the MD5 sum, won't it ?

Sorry for all that mess :/

mseri commented:
There is no need to touch the opam file that is in the 1.0 tarball, opam will use the one from the repository anyways. Just make the url change in the opam file in `packages/rfsm/rfsm.1.0/opam` and add the change to this pull request. You should find the clone of opam repository done by opam-publish for this pull request in ` ~/.opam/plugins/opam-publish/repos/ocaml%opam-repository/`

jserot commented:
> [...] Just make the url change in the opam file in packages/rfsm/rfsm.1.0/opam 

You mean in file `~/.opam/repo/default/packages/rfsm/rfsm.1.0/opam` ?

> and add the change to this pull request

How do i do this ?

mseri commented:
> > [...] Just make the url change in the opam file in packages/rfsm/rfsm.1.0/opam
> 
> You mean in file `~/.opam/repo/default/packages/rfsm/rfsm.1.0/opam` ?

No: `~/.opam/plugins/opam-publish/repos/ocaml%opam-repository/packages/rfsm/rfsm.1.0/opam`
(it should be the place that opam-publish used to clone the opam-repository, otherwise try to remember where it was cloned. If you cannot find it you can clone the repository separately, but let's not worry about it right now).

> > and add the change to this pull request
> 
> How do i do this ?

Enter in the folder, I am assuming bash/zsh here:
```
$ cd ~/.opam/plugins/opam-publish/repos/ocaml%opam-repository/
```

Check that you are in the `opam-publish-lascar.0.5` git branch
```
$ git branch
[... there can be other output here ...]
* opam-publish-lascar.0.5
[... there can be other output here ...]
```

If you are not in that branch (you will likely be in `master`), check it out:
```
git checkout opam-publish-lascar.0.5
```

Update the branch with my changes:
```
git pull
```

Modify `packages/rfsm/rfsm.1.0/opam` (i.e. `~/.opam/plugins/opam-publish/repos/ocaml%opam-repository/packages/rfsm/rfsm.1.0/opam`) with the necessary changes, save the changes, then commit the changes:
```
git add ~/.opam/plugins/opam-publish/repos/ocaml%opam-repository/packages/rfsm/rfsm.1.0/opam
git commit -m "rfsm.1.0: update url"
```

Push the changes
```
git push
```

Or if it does not work:
```
git push user opam-publish-lascar.0.5
```

jserot commented:
Done:

```
$ cd plugins/opam-publish/repos/ocaml%opam-repository/
$ ls
CHANGES.md	COPYING		packages	version
CONTRIBUTING.md	README.md	repo
$ git branch
* master
$ git checkout opam-publish-lascar.0.5
M	packages/rfsm/rfsm.1.0/opam
Branch opam-publish-lascar.0.5 set up to track remote branch opam-publish-lascar.0.5 from user.
Switched to a new branch 'opam-publish-lascar.0.5'
$ git branch
  master
* opam-publish-lascar.0.5
$ git pull
Warning: Permanently added the RSA host key for IP address '...' to the list of known hosts.
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Total 6 (delta 3), reused 3 (delta 3), pack-reused 3
Unpacking objects: 100% (6/6), done.
From github.com:jserot/opam-repository
   54bc35e19a..59d596bc1e  opam-publish-lascar.0.5 -> user/opam-publish-lascar.0.5
Updating 54bc35e19a..59d596bc1e
Fast-forward
 packages/lascar/lascar.0.5/opam | 4 ++++
 1 file changed, 4 insertions(+)
$ vi packages/rfsm/rfsm.1.0/opam
$ tail packages/rfsm/rfsm.1.0/opam
...
synopsis:
  "A toolset for describing and simulating StateChart-like state diagrams."
url {
  src: "http://cloud.ip.univ-bpclermont.fr/~serot/rfsm/dist/rfsm-1.0.tar.gz"
  checksum: "md5=e4d7f1086eb346e7a61760036f059bc6"
}
$ pwd
/Users/jserot/.opam/plugins/opam-publish/repos/ocaml%opam-repository
$ git add packages/rfsm/rfsm.1.0/opam
$ git commit -m "rfsm.1.0: updated url"
[opam-publish-lascar.0.5 d6ba3992df] rfsm.1.0: updated url
 1 file changed, 1 insertion(+), 1 deletion(-)
$ git push
Counting objects: 6, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (4/4), done.
Writing objects: 100% (6/6), 455 bytes | 455.00 KiB/s, done.
Total 6 (delta 3), reused 0 (delta 0)
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To github.com:jserot/opam-repository
   59d596bc1e..d6ba3992df  opam-publish-lascar.0.5 -> opam-publish-lascar.0.5
$ 

```

mseri commented:
Thanks for the help and the patience. The two failures in the ci are: expected camlp4 failure on the system switch, sandboxing restriction on osx. All fine

jserot commented:
Thanks for _your_ help and patience !
